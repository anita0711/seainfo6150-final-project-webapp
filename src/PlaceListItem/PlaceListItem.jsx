import React from "react";
import PlaceImage from "../PlaceImage/PlaceImage";
import styles from "./PlaceListItem.module.css";

const PlaceListItem = (props) => {
  return (
    <div className={styles.placeListItem}>
      <PlaceImage
        img1={props.img1}
        img2={props.img2}
        img3={props.img3}
        img4={props.img4}
      />
      <p>{props.title}</p>
    </div>
  );
};

export default PlaceListItem;
