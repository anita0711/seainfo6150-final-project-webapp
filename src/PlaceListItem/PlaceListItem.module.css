.placeListItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-decoration: none;
}

.placeName {
  color: black;
  background-color: white;
  font-size: large;
  border-radius: 10px;
}

.placeName:hover {
  color: white;
  background-color: darkgray;
}
