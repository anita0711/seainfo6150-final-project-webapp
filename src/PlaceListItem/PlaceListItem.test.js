import React from "react";
import { render } from "@testing-library/react";
import PlaceListItem from "./PlaceListItem.jsx";

it("renders the PlaceListItem component correctly", () => {
  const { container } = render(
    <PlaceListItem
      img1="https://mowryjournal.com/wp-content/uploads/2014/10/grand-teton-np-34-sunrise.jpg"
      title="Grand Teton National Park"
    />
  );
  expect(container).toMatchSnapshot();
});
