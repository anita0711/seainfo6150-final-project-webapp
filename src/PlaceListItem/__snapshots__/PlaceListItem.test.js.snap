// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PlaceListItem component correctly 1`] = `
<div>
  <div
    class="placeListItem"
  >
    <img
      alt="Grand Teton National Park"
      class="image"
      src="https://mowryjournal.com/wp-content/uploads/2014/10/grand-teton-np-34-sunrise.jpg"
    />
    <p
      class="placeName"
    >
      Grand Teton National Park
    </p>
  </div>
</div>
`;
